ext/RMagick/rmfill.c

```diff
@@ -600,7 +601,7 @@ GradientFill_fill(VALUE self, VALUE image_obj)
     double x1, y1, x2, y2;          // points on the line
 
     Data_Get_Struct(self, rm_GradientFill, fill);
-    image = rm_check_destroyed(image_obj);
+    image = rm_check_destroyed(rm_cur_image(image_obj));
```

Comment:
I'm not familiar with the convention in RMagick, but should these methods operate on the entire list instead of just the current image? It seems like this behavior might be confusing. People can still pass in an individual image if they only want to modify one.

Reply:
There are already methods that accept ImageList.
And it becomes confusing (for me, at least).

If there are no clear rules,
I think it's less confusing with handling Image and ImageList both at argument.

Reply:
@Watson1978 I'm not saying it shouldn't accept an ImageList. I'm wondering if it should operate on the *entire* ImageList, and not just the current image.

Reply:
@mockdeep Thank you for your response. OK, I will think about this again at the weekend.